src/Task/FilePermissionTask.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Accompli\Task;
+
+use Accompli\AccompliEvents;
+use Accompli\EventDispatcher\Event\InstallReleaseEvent;
+use Accompli\EventDispatcher\Event\LogEvent;
+use Accompli\EventDispatcher\EventDispatcherInterface;
+use Accompli\Exception\TaskCommandExecutionException;
+use GisoStallenberg\FilePermissionCalculator\FilePermissionCalculator;
+use Psr\Log\LogLevel;
+
+/**
+ * FilePermissionTask.
+ *
+ * @author Deborah van der Vegt <deborah@connectholland.nl>
+ */
+class FilePermissionTask extends AbstractConnectedTask
+{
+    /**
+     * The array with paths to set the permissions in.
+     *
+     * @var array
+     */
+    private $paths;
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            AccompliEvents::INSTALL_RELEASE => array(
+                array('onInstallReleaseUpdateFilePermissions', 0),
+            ),
+        );
+    }
+
+    /**
+     * Constructs a new FilePermissionTask.
+     *
+     * @param array $paths
+     */
+    public function __construct($paths)
+    {
+        $this->paths = $paths;
+    }
+
+    /**
+     * Sets the correct permissions and group for the configured path.
+     *
+     * @param InstallReleaseEvent      $event
+     * @param string                   $eventName
+     * @param EventDispatcherInterface $eventDispatcher
+     */
+    public function onInstallReleaseUpdateFilePermissions(InstallReleaseEvent $event, $eventName, EventDispatcherInterface $eventDispatcher)
+    {
+        $host = $event->getRelease()->getWorkspace()->getHost();
+        $connection = $this->ensureConnection($host);
+
+        $eventDispatcher->dispatch(AccompliEvents::LOG, new LogEvent(LogLevel::INFO, 'Set correct permissions for configured paths...', $eventName, $this, array('event.task.action' => TaskInterface::ACTION_IN_PROGRESS)));
+
+        if ($connection->isConnected() === true && $connection->isDirectory($event->getRelease()->getPath())) {
+            foreach (array_keys($this->paths) as $configuredPath) {
```

Comment:
Might be worth moving all the code within the `foreach` as there seems to be some buggy code within the (refactored?) methods. You can refactor it after writing tests.